edgemesh/pkg/server/dns.go

```diff
@@ -0,0 +1,399 @@
+	}
+	return false
+}
+
+//getHeader get dns pkg head
+func (h *dnsHeader)getHeader(req []byte) {
+	h.transactionID = binary.BigEndian.Uint16(req[0:2])
+	h.flags = binary.BigEndian.Uint16(req[2:4])
+	h.queNum = binary.BigEndian.Uint16(req[4:6])
+	h.ansNum = binary.BigEndian.Uint16(req[6:8])
+	h.authNum = binary.BigEndian.Uint16(req[8:10])
+	h.additNum = binary.BigEndian.Uint16(req[10:12])
+}
+
+//getQuestion get dns questions
+func (q dnsQs)getQuestion(req []byte, offset uint16, head *dnsHeader)  {
+	ost := offset
+	qNum := uint16(len(q))
+
+	for i:=uint16(0); i< qNum ; i++ {
+		tmp := ost
+		ost = q[i].getQName(req,ost)
+		q[i].qType = binary.BigEndian.Uint16(req[ost: ost + twoByteSize])
+		ost += twoByteSize
+		q[i].qClasss = binary.BigEndian.Uint16(req[ost: ost + twoByteSize])
+		ost += twoByteSize
+		q[i].head = head
+		q[i].queByte = req[tmp: ost]
+	}
+}
+
+//getAnswer Generate Answer for the dns question
+func (d *dnsAnswer)getAnswer() (ans []byte) {
+	ans = make([]byte,0)
+
+	if d.qType == aRecord {
+		ans = append(ans,0xc0)
+		ans = append(ans,0x0c)
+
+		tmp16 := make([]byte,2)
+		tmp32 := make([]byte,4)
+
+		binary.BigEndian.PutUint16(tmp16,d.qType)
+		ans = append(ans,tmp16...)
+		binary.BigEndian.PutUint16(tmp16, d.qClass)
+		ans = append(ans,tmp16...)
+		binary.BigEndian.PutUint32(tmp32, d.ttl)
+		ans = append(ans,tmp32...)
+		binary.BigEndian.PutUint16(tmp16, d.dataLen)
+		ans = append(ans,tmp16...)
+		ans = append(ans,d.addr...)
+	}
+
+	return ans
+}
+
+// getQName get dns question domain name
+func (q *dnsQuestion)getQName(req []byte, offset uint16) (uint16) {
+	ost := offset
+
+	for {
+		qbyte := uint16(req[ost])
+
+		if qbyte == 0x00 {
+			q.name = q.name[: uint16(len(q.name))- oneByteSize]
+			return ost + oneByteSize
+		}
+		ost += oneByteSize
+		q.name = append(q.name, req[ost: ost+qbyte]...)
+		q.name = append(q.name,0x2e)
+		ost += qbyte
+	}
+}
+
+// lookupFromMetaManager implement confirm the service exists
+// for earlier version returns true by default
+func lookupFromMetaManager(name string) (exist bool ,err error){
+	return true, nil
+}
+
+// getfromRealDNS returns the dns response from the real DNS server
+func getfromRealDNS(req []byte) []byte {
+	ips := parseNameServer()
+
+	rsp := make([]byte,0)
+	laddr := &net.UDPAddr{
+		IP: net.IPv4zero,
+		Port: 0,
+	}
+	for _, ip := range ips  { // get from real
+		raddr := &net.UDPAddr{
+			IP: ip,
+			Port: 53,
+		}
+		conn, err := net.DialUDP("udp",laddr,raddr)
+		defer conn.Close()
+		if err != nil {
+			continue
+		}
+
+		_, err = conn.Write(req)
+		if err != nil {
+			continue
+		}
+
+		var n int
+		buf := make([]byte,bufSize)
+		n, err = conn.Read(buf)
+		if err != nil {
+			continue
+		}
+
+		if n >0 {
+			rsp = append(rsp,buf[: n]...)
+		}
+	}
+	return rsp
+}
+
+// parseNameServer gets the nameserver from the resolv.conf
+func parseNameServer()([]net.IP){
+	file, err := os.Open("./etc/resolv.conf")
+	if err != nil {
+		glog.Errorf("the resolv.conf has sone error : %s")
```

Comment:
"the resolv.conf has sone error : %s"  -> glog.Errorf("error opening /etc/resolv.conf : %v, err)

Reply:
k